### ISSUE TITLE
bug: `ready.d`  initialisation script not working the same has `docker-entrypoint-initaws.d`

### ISSUE THREAD

**Issue by ntedgi** [localstack/localstack#7289]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

how to reproduce?

using a simple bash init script as configMap and mounts the file to the directory 

`init.sh`

   ```
       #!/bin/bash
          awslocal sqs create-queue --queue-name my-queue
```
the script running when mounting to `/docker-entrypoint-initaws.d`
not running when mounting to `/etc/localstack/init/ready.d/init.sh`

environment: ArgoCD with helm chart

### Expected Behavior

from the docs https://docs.localstack.cloud/references/init-hooks/
both of them should behave the same :

```
        ├── ready.d          <-- executed when localstack becomes ready (currently equivalent to `/docker-entrypoint-initaws.d`)
```

### How are you starting LocalStack?

With the `localstack` script

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    awslocal s3 mb s3://mybucket

### Environment

```markdown
- LocalStack: latest 
- ArgoCd 
- Helm Charts
```

### Anything else?

note: could be related to file permission maybe config-maps mounting not adding execution permission (chmod +x))

**Comment by alexrashed** [comment]
Hi @ntedgi!
Could you please share the logs of the LocalStack pod?
How does your environment look like? You mentioned you are running LocalStack within a Kubernetes cluster, but you also mentioned that you are starting localstack with `docker run`...
The following commands are working for me:
```
$ mkdir ready.d
$ cat <<EOT >> ready.d/init.sh
#!/bin/bash
awslocal sqs create-queue --queue-name my-queue
echo "executed!"
EOT
$ chmod +x ready.d/init.sh
$ docker run --rm -v $PWD/ready.d:/etc/localstack/init/ready.d localstack/localstack
...
2022-12-06T10:27:06.910  INFO --- [  MainThread] localstack.utils.bootstrap : Execution of "start_runtime_components" took 603.02ms
Ready.
2022-12-06T10:27:07.539  INFO --- [   asgi_gw_0] localstack.request.aws     : AWS sqs.CreateQueue => 200
{
    "QueueUrl": "http://localhost:4566/000000000000/my-queue"
}
executed!
```

**Comment by alexrashed** [comment]
I'll close this issue for now. Please feel free to comment here or create a new issue if you are still experiencing this issue.

**Comment by stzu** [comment]
I had a similar problem and was happy to find this issue, only to discover there was no solution to it. Since I've resolve my problem, I'll leave a short description for others to find.

The debug logs in localstack showed me first that my init.sh script was discovered, but then it failed to execute it:
```
2022-12-29T09:15:50.887 DEBUG --- [  MainThread] localstack.runtime.init    : Init scripts discovered: {BOOT: [], START: [], READY: [Script(path='/etc/localstack/init/ready.d/init.sh', stage=READY, state=UNKNOWN)], SHUTDOWN: []}

[...some other logs...]

2022-12-29T09:15:52.732125606Z 2022-12-29T09:15:52.732 DEBUG --- [  MainThread] localstack.runtime.init    : Running READY script /etc/localstack/init/ready.d/init.sh
2022-12-29T09:15:52.746223723Z 2022-12-29T09:15:52.735 ERROR --- [  MainThread] localstack.runtime.init    : Error while running script Script(path='/etc/localstack/init/ready.d/init.sh', stage=READY, state=ERROR)
2022-12-29T09:15:52.746247893Z Traceback (most recent call last):
2022-12-29T09:15:52.746251236Z   File "/opt/code/localstack/localstack/runtime/init.py", line 136, in run_stage
2022-12-29T09:15:52.746253013Z     runner.run(script.path)
2022-12-29T09:15:52.746254550Z   File "/opt/code/localstack/localstack/runtime/init.py", line 69, in run
2022-12-29T09:15:52.746256275Z     exit_code = subprocess.call(args=[], executable=path)
2022-12-29T09:15:52.746257824Z   File "/usr/local/lib/python3.10/subprocess.py", line 345, in call
2022-12-29T09:15:52.746259438Z     with Popen(*popenargs, **kwargs) as p:
2022-12-29T09:15:52.746260926Z   File "/usr/local/lib/python3.10/subprocess.py", line 971, in __init__
2022-12-29T09:15:52.746262528Z     self._execute_child(args, executable, preexec_fn, close_fds,
2022-12-29T09:15:52.746264065Z   File "/usr/local/lib/python3.10/subprocess.py", line 1847, in _execute_child
2022-12-29T09:15:52.746265659Z     raise child_exception_type(errno_num, err_msg, err_filename)
2022-12-29T09:15:52.746267190Z FileNotFoundError: [Errno 2] No such file or directory: '/etc/localstack/init/ready.d/init.sh'
```

As my Docker host is Windows, the init.sh script had CRLF line endings. **After manually changing to LF line endings, the script was properly executed in the container.**

**Comment by Nikola-Milovic** [comment]
@alexrashed I am scratching my head over this, seems like a simple enough setup

```
  localstack:
    container_name: "${LOCALSTACK_DOCKER_NAME-localstack_main}"
    image: localstack/localstack
    ports:
      - "127.0.0.1:4566:4566"
      - "127.0.0.1:4510-4559:4510-4559"
    environment:
      - SERVICES=s3
      - DEBUG=1
      - DOCKER_HOST=unix:///var/run/docker.sock
    volumes:
      - "${LOCALSTACK_VOLUME_DIR:-./.localstack}:/var/lib/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
      - "./tools/docker/scripts/localstack-ready.sh:/etc/localstack/init/ready.d/localstack-ready.sh"
```

Startup indicates its unable to find it

```
 ⠿ Container localstack_main  Created                                                                                    0.0s
Attaching to localstack_main
localstack_main  | Waiting for all LocalStack services to be ready
localstack_main  | 2023-02-21 07:00:28,791 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to run as root, you can set user=root in the config file to avoid this message.
localstack_main  | 2023-02-21 07:00:28,797 INFO supervisord started with pid 14
localstack_main  | 2023-02-21 07:00:29,801 INFO spawned: 'dashboard' with pid 20
localstack_main  | 2023-02-21 07:00:29,803 INFO spawned: 'infra' with pid 21
localstack_main  | 2023-02-21 07:00:29,807 INFO success: dashboard entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
localstack_main  | 2023-02-21 07:00:29,808 INFO exited: dashboard (exit status 0; expected)
localstack_main  | (. .venv/bin/activate; exec bin/localstack start --host)
localstack_main  | Starting local dev environment. CTRL-C to quit.
localstack_main  | 2023-02-21 07:00:31,584 INFO success: infra entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
localstack_main  |
localstack_main  | LocalStack version: 0.12.14
localstack_main  | LocalStack Docker container id: 03c4b10a6ef6
localstack_main  | LocalStack build date: 2021-07-01
localstack_main  | LocalStack build git hash: aeff8862
localstack_main  |
localstack_main  | Starting DNS servers (tcp/udp port 53 on 0.0.0.0)...
localstack_main  | Starting edge router (https port 4566)...
localstack_main  | Starting mock S3 service on http port 4566 ...
localstack_main  | 2023-02-21T07:00:34:DEBUG:bootstrap.py: Loading plugins - scope "services", module "localstack": <function register_localstack_plugins at 0x7f38b0270a70>
localstack_main  | 2023-02-21T07:00:35:INFO:localstack.multiserver: Starting multi API server process on port 32977
localstack_main  | [2023-02-21 07:00:35 +0000] [22] [INFO] Running on https://0.0.0.0:4566 (CTRL + C to quit)
localstack_main  | 2023-02-21T07:00:35:INFO:hypercorn.error: Running on https://0.0.0.0:4566 (CTRL + C to quit)
localstack_main  | [2023-02-21 07:00:35 +0000] [22] [INFO] Running on http://0.0.0.0:32977 (CTRL + C to quit)
localstack_main  | 2023-02-21T07:00:35:INFO:hypercorn.error: Running on http://0.0.0.0:32977 (CTRL + C to quit)
localstack_main  | Waiting for all LocalStack services to be ready
localstack_main  | 2023-02-21 07:00:36,237:API:  * Running on all addresses.
localstack_main  |    WARNING: This is a development server. Do not use it in a production deployment.
localstack_main  | 2023-02-21 07:00:36,241:API:  * Running on http://172.21.0.2:39877/ (Press CTRL+C to quit)
localstack_main  | Waiting for all LocalStack services to be ready
localstack_main  | Ready.
localstack_main  | 2023-02-21 07:00:45,775:API: 127.0.0.1 - - [21/Feb/2023 07:00:45] "GET / HTTP/1.1" 200 -
localstack_main  | 2023-02-21T07:00:45:INFO:localstack.utils.analytics.profiler: Execution of "start_api_services" took 10626.45ms
```

But the weirdest part is the owner of the script is `node`, while on my system it's the user

```
/etc/localstack/init/ready.d # ls -la
total 12
drwxr-xr-x    2 root     root          4096 Feb 21 06:54 .
drwxr-xr-x    3 root     root          4096 Feb 21 06:54 ..
-rwxrwxr-x    1 node     node            63 Feb 21 07:00 localstack-ready.sh
```

```
rwxrwxr-x 1 nikola nikola   63 феб 21 08:00 localstack-ready.sh
```

Have you ever encountered something like this? I am starting up only this service from the `docker-compose` file, not sure where the node user is coming from, maybe that is causing the issue?

**Comment by alexrashed** [comment]
Hi @Nikola-Milovic!
I just tested your setup and it worked just fine for me. Your log output also doesn't really match the docker compose file you posted, or it's stripped down?
Nevertheless, it's not weird at all that the script is owned by the node user, because user ownerships are defined by user IDs. "ls" translates these user IDs to usernames. On your system, your user has UID `x`, but in the localstack image, this user ID just corresponds to the user `node`.

**Comment by Nikola-Milovic** [comment]
@alexrashed thanks for the swift response. My image was outdated... That was the root issue

**Comment by glaubermd-leaf** [comment]
I've tested on localstack:latest Docker image and it's not working as expected. It works on localstack:1.4 though.

By running a tail on the container logs, I can see it found the startup scripts only when running image version 1.4:

```
$ docker-compose logs -f|egrep "(READY|ERROR|Init scripts discovered|ready\.d)"

localstack-1_4-local | 2023-09-12T22:43:32.022 DEBUG --- [  MainThread] localstack.runtime.init    : Init scripts discovered: {BOOT: [], START: [], READY: [Script(path='/etc/localstack/init/ready.d/01-dynamodb.sh', stage=READY, state=UNKNOWN)], SHUTDOWN: []}
localstack-1_4-local | 2023-09-12T22:43:33.247 DEBUG --- [  MainThread] localstack.runtime.init    : Running READY script /etc/localstack/init/ready.d/01-dynamodb.sh
```

**Comment by alexrashed** [comment]
Hi @glaubermd-leaf!
I just verified that the init scripts work just fine with the latest image for me.
Unfortunately, it's a bit hard to follow closed issues, and your issue might not be related to this one.
Could you please create a new issue with detailed in formation on:
- How are you starting localstack?
  - If you are using docker-compose, please add your compose file.
  - If you are using our CLI, please add the command you use to start localstack.
- How can you reproduce that it works with version 1.4, but does not with the latest version?
- What file permissions do your files have? (`ls -al`)
Thanks!

**Comment by rafikeefar** [comment]
> I had a similar problem and was happy to find this issue, only to discover there was no solution to it. Since I've resolve my problem, I'll leave a short description for others to find.
> 
> The debug logs in localstack showed me first that my init.sh script was discovered, but then it failed to execute it:
> 
> ```
> 2022-12-29T09:15:50.887 DEBUG --- [  MainThread] localstack.runtime.init    : Init scripts discovered: {BOOT: [], START: [], READY: [Script(path='/etc/localstack/init/ready.d/init.sh', stage=READY, state=UNKNOWN)], SHUTDOWN: []}
> 
> [...some other logs...]
> 
> 2022-12-29T09:15:52.732125606Z 2022-12-29T09:15:52.732 DEBUG --- [  MainThread] localstack.runtime.init    : Running READY script /etc/localstack/init/ready.d/init.sh
> 2022-12-29T09:15:52.746223723Z 2022-12-29T09:15:52.735 ERROR --- [  MainThread] localstack.runtime.init    : Error while running script Script(path='/etc/localstack/init/ready.d/init.sh', stage=READY, state=ERROR)
> 2022-12-29T09:15:52.746247893Z Traceback (most recent call last):
> 2022-12-29T09:15:52.746251236Z   File "/opt/code/localstack/localstack/runtime/init.py", line 136, in run_stage
> 2022-12-29T09:15:52.746253013Z     runner.run(script.path)
> 2022-12-29T09:15:52.746254550Z   File "/opt/code/localstack/localstack/runtime/init.py", line 69, in run
> 2022-12-29T09:15:52.746256275Z     exit_code = subprocess.call(args=[], executable=path)
> 2022-12-29T09:15:52.746257824Z   File "/usr/local/lib/python3.10/subprocess.py", line 345, in call
> 2022-12-29T09:15:52.746259438Z     with Popen(*popenargs, **kwargs) as p:
> 2022-12-29T09:15:52.746260926Z   File "/usr/local/lib/python3.10/subprocess.py", line 971, in __init__
> 2022-12-29T09:15:52.746262528Z     self._execute_child(args, executable, preexec_fn, close_fds,
> 2022-12-29T09:15:52.746264065Z   File "/usr/local/lib/python3.10/subprocess.py", line 1847, in _execute_child
> 2022-12-29T09:15:52.746265659Z     raise child_exception_type(errno_num, err_msg, err_filename)
> 2022-12-29T09:15:52.746267190Z FileNotFoundError: [Errno 2] No such file or directory: '/etc/localstack/init/ready.d/init.sh'
> ```
> 
> As my Docker host is Windows, the init.sh script had CRLF line endings. **After manually changing to LF line endings, the script was properly executed in the container.**

dos2unix 001-init-dynamodb.sh -- this works for me
